cmd/cloud.go

```diff
@@ -343,20 +345,67 @@ func getCmdCloud(gs *state.GlobalState) *cobra.Command {
 	}
 
 	exampleText := getExampleText(gs, `
-  {{.}} cloud script.js`[1:])
+  # Authenticate with Grafana Cloud k6
+  $ {{.}} cloud login
+
+  # Run a k6 script in the Grafana Cloud k6
+  $ {{.}} cloud run script.js
+
+  # Run a k6 archive in the Grafana Cloud k6
+  $ {{.}} cloud run archive.tar
+
+  # [deprecated] Run a k6 script in the Grafana Cloud k6
+  $ {{.}} cloud script.js
+
+  # [deprecated] Run a k6 archive in the Grafana Cloud k6
+  $ {{.}} cloud archive.tar`[1:])
 
 	cloudCmd := &cobra.Command{
 		Use:   "cloud",
 		Short: "Run a test on the cloud",
-		Long: `Run a test on the cloud.
+		Long: `Run a test archive in the Grafana Cloud k6.
 
-This will execute the test on the k6 cloud service. Use "k6 login cloud" to authenticate.`,
-		Example: exampleText,
-		Args:    exactArgsWithMsg(1, "arg should either be \"-\", if reading script from stdin, or a path to a script file"),
+This will execute the test in the Grafana Cloud k6 service. Be sure to run the "k6 cloud login" command prior to
+authenticate with Grafana Cloud k6.`,
+		Args: exactCloudArgs(),
+		Deprecated: `the k6 team is in the process of modifying and deprecating the "k6 cloud" command behavior.
+In the future, the "cloud" command will only display a help text, instead of running tests in the Grafana Cloud k6.
+
+To run tests in the cloud, users are now invited to migrate to the "k6 cloud run" command instead.
+`,
 		PreRunE: c.preRun,
 		RunE:    c.run,
+		Example: exampleText,
 	}
+
+	// Register `k6 cloud` subcommands
+	cloudCmd.AddCommand(getCmdCloudRun(gs))
+	cloudCmd.AddCommand(getCmdCloudLogin(gs))
+
 	cloudCmd.Flags().SortFlags = false
 	cloudCmd.Flags().AddFlagSet(c.flagSet())
+
 	return cloudCmd
 }
+
+func exactCloudArgs() cobra.PositionalArgs {
+	return func(_ *cobra.Command, args []string) error {
+		const baseErrMsg = "the k6 cloud command expects either a subcommand such as `run` or `login`, or " +
+			"a single argument consisting in a path to a script/archive, or the `-` symbol instructing " +
+			"the command to read the test content from stdin"
+
+		if len(args) == 0 {
+			return fmt.Errorf(baseErrMsg + "; " + "received no arguments")
+		}
+
+		hasSubcommand := len(args) >= 1 && (args[0] == "run" || args[0] == "login")
+		if !hasSubcommand {
+			return fmt.Errorf(
+				baseErrMsg+"; "+"received %d arguments %q, and none of them is a subcommand",
+				len(args), strings.Join(args, " "),
```

Comment:
I think the way below is a bit more accurate, as we only care about the first argument (to be a valid subcommand):

```suggestion
				baseErrMsg+"; "+"received %d arguments %q, and %s is not a valid subcommand",
				len(args), strings.Join(args, " "), args[0],
```